### PR TITLE
Bids 2487/slot based exporter

### PR DESCRIPTION
**Pull request by peterbitfly**
### <samp>🤖 Generated by Copilot at ee487ee</samp>

This pull request implements the slot exporter feature, which exports and saves the slot, epoch, block, and validator data to Bigtable and the database more efficiently and reliably. It also removes or updates the code and configuration related to the previous export features, the epoch write cache, and the network parameters. It affects the files `cmd/misc/main.go`, `cmd/explorer/main.go`, `db/bigtable.go`, `db/db.go`, `db/migrations/20231009124756_add_blocks_finalized_column.sql`, `exporter/slot_exporter.go`, `local-deployment/docker-compose.yml`, `local-deployment/network-params.json`, `rpc/interfaces.go`, `rpc/lighthouse.go`, `services/services.go`, and `types/config.go`, and `types/exporter.go`.

### CODE REVIEW

**Review by D13ce — CHANGES_REQUESTED** [review]

File: exporter/slot_exporter.go [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+	err = db.BigtableClient.SaveProposal(block)
+	if err != nil {
+		return fmt.Errorf("error exporting proposal to bigtable for slot %v: %w", block.Slot, err)
+	}
+
+	// save the block data to the db
+	err = db.SaveBlock(block, false)
+	if err != nil {
+		logger.Errorf("error saving slot to the db: %w", err)
+	}
```
[the remark this answers is not in the document]
Comment (peterbitfly) [reply exporter/slot_exporter.go:338]: saving sync & proposal duties don't take a lot of time, so parallelism won't help much in this case. `db.SaveBlock` should only be called once the bigtable writes have been completed to ensure consistency.

File: services/services.go [hunk 2]
```diff
@@ -790,7 +790,7 @@ func getIndexPageData() (*types.IndexPageData, error) {
 
 	latestFinalizedEpoch := LatestFinalizedEpoch()
 	var epochs []*types.IndexPageDataEpochs
-	err = db.ReaderDb.Select(&epochs, `SELECT epoch, (epoch <= $1) AS finalized , eligibleether, globalparticipationrate, votedether FROM epochs ORDER BY epochs DESC LIMIT 15`, latestFinalizedEpoch)
```
[the remark this answers is not in the document]
Comment (peterbitfly) [reply services/services.go:793]: We should do that in a separate BIDS

[the remark this answers is not in the document]
Comment (D13ce) [reply services/services.go:793]: Created

File: exporter/slot_exporter.go [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+						return err
+					}
+				}
+			}
+		} else { // check if a late slot has been proposed in the meantime
+			if len(dbSlot.BlockRoot) < 32 && header != nil { // we have no slot in the db, but the node has a slot, export it
+				err := ExportSlot(client, dbSlot.Slot, utils.EpochOfSlot(dbSlot.Slot) == head.HeadEpoch)
+				if err != nil {
+					return fmt.Errorf("error exporting slot %v: %w", dbSlot.Slot, err)
+				}
```
[the remark this answers is not in the document]
Comment (D13ce) [reply exporter/slot_exporter.go:168]: I did not mean "a log entry identical to this". Please adapt.

**Review by D13ce — APPROVED** [review]
utACK :tada:

**Pull request merged**